app/models/post.rb

```diff
@@ -1,3 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
👏 Good job on your recent changes! However, there's a small enhancement that can improve your code:

- [ ] **Using Rails Active Callbacks:** When creating a new post, it's expected to automatically call `update_posts_counter` and update the post count for the author. A cleaner way to achieve this is by using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html), specifically `after_save`. This callback ensures that the `update_posts_counter` method is called every time a new post is saved to the database.
- [ ] [Optional] **Making the Method Private:** Since the `update_posts_counter` method is intended for internal use within the class, it's good practice to make it private. This prevents it from being called outside the class.

    ```ruby
    class Post < ApplicationRecord
      # ... (existing code)

      private

      def update_posts_counter
        # ... (existing implementation)
      end
    end
    ```

By incorporating these changes, you'll follow best practices and ensure a clean and maintainable codebase. Keep up the good work! 🚀🛠️